src/Nethermind/Nethermind.Api/Extensions/IConsensusPlugin.cs

```diff
@@ -24,8 +25,10 @@ namespace Nethermind.Api.Extensions
 {
     public interface IConsensusPlugin : INethermindPlugin
     {
-        Task<IBlockProducer> InitBlockProducer(ITxSource? txSource = null);
+        Task<IBlockProducer> InitBlockProducer(IBlockProductionTrigger? blockProductionTrigger = null, ITxSource? additionalTxSource = null);
```

Comment:
Ad

Reply:
It is dealt inside via composite pattern.

So we want the consensus to be in charge of how it adds transactiosn to blocks, as there are cases where there are consensus transactions that need to go first into the block (AuRa, POSDAO), but we also want to be able to inject some tx's from the outside (MEV), so I was debating 2 options:
1. additionalTxSource (named like that in order to not confuse that its a main tx source)
2. event to inject additional transactions

went with 1.